src/Calculator/Views/CalculatorProgrammerBitFlipPanel.xaml.cpp

```diff
@@ -185,35 +174,51 @@ void CalculatorProgrammerBitFlipPanel::UpdateCheckedStates()
         return;
     }
 
-    static const wchar_t ch0 = LocalizationSettings::GetInstance().GetDigitSymbolFromEnUsDigit(L'0');
-
-    // Filter any unwanted characters from the displayed string.
-    static constexpr array<wchar_t, 4> unwantedChars = { L' ', Utils::LRE, Utils::PDF, Utils::LRO };
-
-    wstringstream stream;
-    wstring displayValue = Model->BinaryDisplayValue->Data();
-    for (const wchar_t& c : displayValue)
+    m_updatingCheckedStates = true;
+    unsigned int bitIndex = 0;
+    auto it = m_flipButtons.begin();
+    for (bool val : Model->BinaryDigits)
     {
-        if (find(begin(unwantedChars), end(unwantedChars), c) == unwantedChars.end())
+        // Highest bit (64) is at index 0 in bit string.
+        // To get bit 0, grab from opposite end of string.
+        auto checkbox = *it;
+        if (forceUpdate || checkbox->IsChecked->Value != val)
         {
-            stream << c;
+            checkbox->IsChecked = val;
+            // Only generate the string when the value changed
+            checkbox->SetValue(AutomationProperties::NameProperty, GenerateAutomationPropertiesName(bitIndex, val));
         }
+        ++bitIndex;
+        ++it;
     }
 
-    wstring rawDisplay = stream.str();
-    size_t paddingCount = s_numBits - rawDisplay.length();
-    wstring setBits = wstring(paddingCount, ch0) + rawDisplay;
-    assert(setBits.length() == s_numBits);
+    m_updatingCheckedStates = false;
+}
 
-    m_updatingCheckedStates = true;
-    for (unsigned int bitIndex = 0; bitIndex < s_numBits; bitIndex++)
+bool CalculatorProgrammerBitFlipPanel::ShouldEnableBit(BitLength length, int index)
+{
+    switch (length)
     {
-        // Highest bit (64) is at index 0 in bit string.
-        // To get bit 0, grab from opposite end of string.
-        wchar_t bit = setBits[s_numBits - bitIndex - 1];
-
-        m_flipButtons[bitIndex]->IsChecked = (bit != ch0);
+    case BitLength::BitLengthQWord:
+        return index <= 63;
+    case BitLength::BitLengthDWord:
+        return index <= 31;
+    case BitLength::BitLengthWord:
+        return index <= 15;
+    case BitLength::BitLengthByte:
+        return index <= 7;
     }
+    return false;
+}
 
-    m_updatingCheckedStates = false;
+String ^ CalculatorProgrammerBitFlipPanel::GenerateAutomationPropertiesName(int position, bool value) const
+{
+    auto resourceLoader = AppResourceProvider::GetInstance();
+
+    String ^ indexName = resourceLoader.GetResourceString(ref new Platform::String(to_wstring(position).c_str()));
+    String ^ bitName = resourceLoader.GetResourceString(L"BitAutomationName");
+    String ^ valueName = resourceLoader.GetResourceString(L"ValueAutomationName");
+
+    return indexName + bitName + valueName
```

Comment:
>indexName + bitName + valueName [](start = 11, length = 31)

I think automation name gets localized.  As such, I don't think you want to be concatenating in this way.

Reply:
Looks like the converter was doing this the same way before your change.  We can probably leave this as is and track this as a separate issue if you don't fix this here.

Reply:
Yes. But you are right, we can improve this part. I will update the PR tonight